homeassistant/components/mqtt/vacuum/__init__.py

```diff
@@ -24,16 +31,56 @@
     async_setup_entity_state,
 )
 
+_LOGGER = logging.getLogger(__name__)
+
+MQTT_VACUUM_DOCS_URL = "https://www.home-assistant.io/integrations/vacuum.mqtt/"
+
+
+# The legacy schema for MQTT vacuum was deprecated with HA Core 2023.7.0
+# and is will be removed with HA Core 2024.2.0
+def warn_for_deprecation_legacy_schema(
+    hass: HomeAssistant, config: ConfigType, discovery_data: DiscoveryInfoType | None
+) -> None:
+    """Warn for deprecation of legacy schema."""
+    if config[CONF_SCHEMA] == STATE:
+        return
+
+    key_suffix = "yaml" if discovery_data is None else "discovery"
+    translation_key = f"deprecation_mqtt_legacy_vacuum_{key_suffix}"
+    async_create_issue(
+        hass,
+        DOMAIN,
+        translation_key,
+        breaks_in_ha_version="2024.2.0",
+        is_fixable=False,
+        is_persistent=True,
```

Comment:
Why do we set this to persistent? Shouldn't it be non-persistent, so the issue goes away after a restart when the user has updated the config?